Mono.Addins/Mono.Addins/RuntimeAddin.cs

```diff
@@ -321,54 +321,41 @@ public Type GetType (string typeName, bool throwIfNotFound)
 
 		internal Type GetType (string typeName, string assemblyName, bool throwIfNotFound)
 		{
-			EnsureAssembliesLoaded ();
-
-			// Look in the addin assemblies and in dependent add-ins.
-			// PERF: Unrolled from GetAllAssemblies and GetAllDependencies to avoid allocations.
-			Type at = (string.IsNullOrEmpty (assemblyName) ? Type.GetType (typeName, false) : null)
-				?? GetTypeRecursive (typeName, assemblyName);
-			if (at != null)
-				return at;
+			// Try looking in Mono.Addins without loading the addin assemblies.
+			var type = (string.IsNullOrEmpty (assemblyName) ? Type.GetType (typeName, false) : null)
```

Comment:
This should be the first big speed boost, since we're only going to look in Mono.Addins assembly _once_, not for every addin and their dependency.